File: Data Structures & Algorithms/is-anagram/submission-1.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False 
        sd = {}
        td = {}
        for cs in s:
            sd[cs] = 1 if cs not in sd.keys() else sd[cs] + 1
        for ct in t:
            td[ct] = 1 if ct not in td.keys() else td[ct] + 1
        return sd == td
        
          

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> mp;
        for (int n : nums) {
            mp[n] += 1;
        }
        vector<vector<int>> buckets (nums.size() + 1);
        for (pair<int, int> p : mp) {
            buckets[p.second].push_back(p.first);
        }
        vector<int> result; 
        for (int i = nums.size(); i >= 0; --i) {
            for (int num : buckets[i]) {
                result.push_back(num);
                if (result.size() == k) {
                    return result;
                }
            }
        }
    }
};
